src/pages/iou/ReceiptSelector/index.native.js

```diff
@@ -0,0 +1,314 @@
+
+                    return reject(new Error(`Error during attachment selection: ${response.errorMessage}`));
+                }
+
+                return resolve(response.assets);
+            });
+        });
+
+    const takePhoto = () => {
+        if (!camera.current) {
+            showCameraAlert();
+            return;
+        }
+
+        camera.current
+            .takePhoto({
+                qualityPrioritization: 'speed',
+                flash: flash ? 'on' : 'off',
+            })
+            .then((photo) => {
+                IOU.setMoneyRequestReceipt(`file://${photo.path}`, photo.path);
+                IOU.navigateToNextPage(props.iou, iouType.current, reportID.current, props.report);
+            })
+            .catch(() => {
+                showCameraAlert();
+            });
+    };
+
+    Camera.getCameraPermissionStatus().then((permissionStatus) => {
+        setPermissions(permissionStatus);
+    });
+
+    const cameraComponent = () => (
+        <Camera
+            ref={camera}
+            device={device}
+            style={[styles.cameraView]}
+            isActive
+            photo
+        />
+    );
+
+    const loadingComponent = () => (
+        <View style={[styles.cameraView]}>
+            <ActivityIndicator
+                size="large"
+                style={{flex: 1}}
+                color={themeColors.textSupporting}
+            />
+        </View>
+    );
+
+    const permissionComponent = () => (
+        <View style={[styles.cameraView, styles.permissionView]}>
+            <Hand
+                width={152}
+                height={200}
+                style={{paddingBottom: 20}}
+            />
+            <Text style={[styles.textReceiptUpload]}>{translate('receipt.takePhoto')}</Text>
+            <Text style={[styles.subTextReceiptUpload]}>{translate('receipt.cameraAccess')}</Text>
+            <PressableWithFeedback
+                accessibilityLabel={translate('receipt.givePermission')}
+                accessibilityRole="button"
+            >
+                <Button
+                    medium
+                    success
+                    text={translate('receipt.givePermission')}
+                    style={[styles.p9, styles.pt5]}
+                    onPress={() => {
+                        if (permissions === 'not-determined') {
+                            Camera.requestCameraPermission().then((permissionStatus) => {
+                                setPermissions(permissionStatus);
+                            });
+                        } else {
+                            Linking.openSettings();
+                        }
+                    }}
+                />
+            </PressableWithFeedback>
+        </View>
+    );
+
+    const getCameraView = () => {
+        if (permissions !== CONST.RECEIPT.PERMISSION_AUTHORIZED) {
+            return permissionComponent();
+        }
+        return device == null ? loadingComponent() : cameraComponent();
+    };
```

Comment:
I'm not sure about those inline function components, can we avoid them? Or break the component into multiple small components

Reply:
I removed them